### DISCUSSION TITLE
upgrade piranha version, but how?

### DISCUSSION THREAD

**Discussion by erdemkara** [PiranhaCMS/piranha.core#1847] [General]
I have a Piranha running on .netcore 3.1. 
I want to upgrade it to version 9.2 but I guess there is no healthy way. 
I need to do a fresh install and move all the content to the new install? is it correct? this is very costly.

By the way, I did a fresh install, I impoted my old database backup into the new empty database, but I had a lot of problems.
What is the least costly way to upgrade?
It would be great if you open a title for the documents for this.

**Comment by tidyui** [comment]
What version exactly is your application developed in? As far as the database goes Entity Framework should be able to upgrade the database to the latest version (make a copy of it :)) as it's all migrations. Breakings changes between 8 & 9 should mostly startup related changes (I think). I think that you can't upgrade an SqLite database, this is due to the fact the Microsoft changed the underlying SqLite library between some versions.

Anyway, it would be interesting to here more about your current project and setup as it should be fully possible to upgrade.

**Comment by erdemkara** [comment]
This is really interesting. I'm trying to upgrade from piranha 8.0.1. I had previously developed with the source code of 8.0.2 and now I have created a new application with 9.0.2. Before running the new application, I ran the migrations and upgraded the old database. During this upgrade the home page is automatically deleted. Then I create a new home page myself, but I realized that the slug is not enough to be "home" for the home page to be valid, also the sort order from the database must be 0. Frankly, I couldn't do this upgrade in a healthy way, I'm probably doing something wrong, but I couldn't get out of it. Any idea? Thank you.

**Reply by tidyui** [reply]
We just upgraded our official homepage (https://piranhacms.org) from version `8.4` (`netcore3.1`) to version `10.0` (`net6`) without any issues besides some minor code changes. Did you do any other refactoring while upgrading, for example change namespaces on your content types that would make the system consider them to be different content types (which can cause pages to be removed depending on how your application is set up).
